listenbrainz/webserver/static/js/src/UserArtists.tsx

```diff
@@ -0,0 +1,153 @@
+
+import APIService from "./APIService";
+import Bar from "./Bar";
+import ErrorBoundary from "./ErrorBoundary";
+
+export type UserArtistsData = Array<{
+  id: string;
+  count: number;
+}>;
+
+export type UserArtistsProps = {
+  user: ListenBrainzUser;
+  apiUrl: string;
+};
+
+export type UserArtistsState = {
+  data: UserArtistsData;
+};
+
+export default class UserArtists extends React.Component<
+  UserArtistsProps,
+  UserArtistsState
+> {
+  APIService: APIService;
+
+  private baseUrl: string; // Base URL of artist statistic page
+  private ROWS_PER_PAGE = 25; // Number of atists to be shown on each page
+  private maxListens = 0; // Number of listens for first artist used to scale the graph
+  private totalPages = 0; // Total namber of pages, useful for pagination
+  private currPage = 1; // Current page, 1 by default
+
+  constructor(props: UserArtistsProps) {
+    super(props);
+
+    this.APIService = new APIService(
+      props.apiUrl || `${window.location.origin}/1`
+    ); // Used to access LB API
+
+    this.baseUrl = window.location.origin + window.location.pathname;
+
+    this.state = {
+      data: [],
+    };
+  }
+
+  async componentDidMount() {
+    const { user } = this.props;
+
+    // Fetch page number from URL
+    const url = new URL(window.location.href);
+    if (url.searchParams.get("page")) {
+      this.currPage = Number(url.searchParams.get("page"));
+    } else {
+      this.currPage = 1;
+    }
+
+    // Fetch data from backend
+    const offset = (this.currPage - 1) * this.ROWS_PER_PAGE;
+    try {
+      let data = await this.APIService.getUserStats(
+        user.name,
+        undefined,
+        undefined,
+        1
+      );
+      this.maxListens = data.payload.artists[0].listen_count;
+      this.totalPages = Math.ceil(
+        data.payload.total_artist_count / this.ROWS_PER_PAGE
+      );
+      data = await this.APIService.getUserStats(
+        user.name,
+        undefined,
+        offset,
+        this.ROWS_PER_PAGE
+      );
+      this.setState({ data: this.processData(data, offset) });
+    } catch (error) {
+      // Error Boundaries don't catch errors in async code.
+      // Throwing an error in setState fixes this.
+      // This is a hacky solution but should be fixed with upcoming concurrent mode in React.
+      this.setState(() => {
+        throw error;
+      });
+    }
+  }
+
+  processData = (
+    data: UserArtistsResponse,
+    offset: number
+  ): UserArtistsData => {
+    const result = data.payload.artists
+      .map((elem, idx: number) => {
+        return {
+          id: `${offset + idx + 1}. ${elem.artist_name}`,
+          count: elem.listen_count,
+        };
+      })
+      .reverse();
+    return result;
+  };
+
+  render() {
+    const { data } = this.state;
+    const prevPage = this.currPage - 1;
+    const nextPage = this.currPage + 1;
+
+    return (
+      <div>
+        <div className="row">
+          <div className="col-md-8" style={{ height: "75em" }}>
```

Comment:
Here and for the pager below, why not use the full available width?
It looks a bit odd on a wide screen at the moment.